### PR TITLE
TINY-8259: Change default table settings - `table_style_by_css` and `table_use_colgroups`

### PR DESCRIPTION
**Pull request by ltrouton**
Related Ticket: TINY-8259

Description of Changes:
* Change `table_style_by_css` and `table_use_colgroups` option defaults from `false` to `true`
* Update tests to work with new defaults

Pre-checks:
* [X] Changelog entry added
* [X] Tests have been added (if applicable)
* [X] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [X] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Review by HAFRMO — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md [hunk 1]
```diff
@@ -84,6 +84,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Replaced 'Powered by Tiny' link text with logo #TINY-8371
 - The `mceInsertTable` command can no longer open the insert table dialog. Use the `mceInsertTableDialog` command instead #TINY-8273
 - Moved non-UI table functionality to core #TINY-8273
+- The `table_style_by_css` option now defaults to `true` #TINY-8259
+- The `table_use_colgroups` option now defaults to `true` #TINY-8259
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:88]:
```suggestion
- The `table_style_by_css` option has been set to `true` by default #TINY-8259
- The `table_use_colgroups` option has been set to `true` by default #TINY-8259
```
Maybe to match with other similar changes above? Either that or change the ones above to match this format, as tbh I do prefer what you have.

File: modules/tinymce/src/core/test/ts/browser/table/InsertTableTest.ts [hunk 2]
```diff
@@ -1,38 +1,49 @@
...
-           'background-color,border,padding,border-spacing,border-collapse'
-    },
-    base_url: '/project/tinymce/js/tinymce',
-    statusbar: false
-  }, []);
+  Arr.each([
+    { label: 'no colgroup', withColgroup: false },
+    { label: 'colgroup', withColgroup: true },
+  ], (scenario) => {
+    const { label, withColgroup } = scenario;
```
Comment (lnewson) [inline modules/tinymce/src/core/test/ts/browser/table/InsertTableTest.ts:14]: This is sort of pointless if we're just destructing it immediately without any renaming or similar, so maybe something like this or just use `scenario.label`, etc... below instead?
```suggestion
  ], ({ label, withColgroup }) => {
```

**Review by lnewson — APPROVED** [review]

**Comment by TheSpyder** [comment]
This needs to be rebased after #7583 was force pushed, it's impossible to review now. I'm really thinking we take a serious look at stacked commits (via graphite) after 6.0 is done.

**Review by spocke — APPROVED** [review]
This PR is now a bit messed up I assume that the target branch has been rebased or something so these branches has been diverged. Anyway I assumed that just the two last commits are that should have been part of this PR and that looks fine.

- [x] Rebase this against the target branch

**Comment by TheSpyder** [comment]
I'm converting this to draft until we finish #7583 and can do a final rebase here

File: modules/tinymce/src/core/test/ts/browser/table/InsertTableTest.ts (hunk 2 above)
Comment (ltrouton) [reply modules/tinymce/src/core/test/ts/browser/table/InsertTableTest.ts]: Done

File: modules/tinymce/CHANGELOG.md (hunk 1 above)
Comment (ltrouton) [reply modules/tinymce/CHANGELOG.md]: I have made it the same as the other entries

**Pull request merged**